spring-cloud-aws-sns/src/main/java/io/awspring/cloud/sns/sms/attributes/SmsMessageAttributes.java

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sns.sms.attributes;
+
+import static io.awspring.cloud.sns.sms.attributes.ConvertToMessageAttributes.populateMapWithNumberValue;
+import static io.awspring.cloud.sns.sms.attributes.ConvertToMessageAttributes.populateMapWithStringValue;
+
+import java.util.HashMap;
+import java.util.Map;
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue;
+
+/**
+ * @author Matej Nedic
+ * @since 3.0.0
+ */
+public class SmsMessageAttributes {
+	private String senderID;
+	private String originationNumber;
+	private String maxPrice;
+	private SmsType smsType;
+	private String entityId;
+	private String templateId;
+	private String messageGroupId;
+	private String deduplicationId;
```

Comment:
Deduplication id makes sense only for sending to topics, so it should not be a property on the top level attributes used also for sending to a number. https://docs.aws.amazon.com/sns/latest/dg/fifo-message-dedup.html